src/catch2/catch_test_spec.cpp

```diff
@@ -42,6 +42,9 @@ namespace Catch {
     {}
 
     bool TestSpec::TagPattern::matches( TestCaseInfo const& testCase ) const {
+        if ( m_tag.empty() ) {
+            return testCase.tags.empty();
+        }
```

Comment:
this might deserve a comment since it's really not obvious why we are doing this check